### PR TITLE
add remote config support for sampling rules

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
https://docs.google.com/document/d/12lXVniuaYenanxnpt5gJJ76a1X7gFOfmCYVQVXlHNco/edit?usp=sharing

forcing system test to run with changes in .github/workflows/system-tests.yml, will delete the change after PR passes review

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/config.js
Comment (rochdev) [inline packages/dd-trace/src/config.js:869]: Why is this needed? It looks like cases where it should be a string VS an array are isolated.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/config.js
Comment (ida613) [reply packages/dd-trace/src/config.js:869]: when it is null/undefined, we cannot do sample_rate -> sampleRate mapping on it

File: .github/workflows/system-tests.yml
Comment (simon-id) [inline .github/workflows/system-tests.yml:86]: reminder to remove before merging :+1:

File: packages/dd-trace/src/appsec/remote_config/capabilities.js
Comment (simon-id) [inline packages/dd-trace/src/appsec/remote_config/capabilities.js:19]: you're adding a new capability but you're never using it ?

File: packages/dd-trace/src/config.js
Comment (simon-id) [inline packages/dd-trace/src/config.js:972]: FYI this is equivalent to:
```suggestion
    if (value == null) {
```
up to you to decide which one you prefere

File: packages/dd-trace/src/config.js
Comment (simon-id) [inline packages/dd-trace/src/config.js:977]: we have modern javascript syntax now! enjoy it!
```suggestion
      value = value?.split(',')
```

File: packages/dd-trace/test/config.spec.js
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:1]: i believe there are *at least* two more tests to update here, the one that test for env var, and the one for programmatic config

**Review by simon-id — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/appsec/remote_config/capabilities.js
Comment (ida613) [reply packages/dd-trace/src/appsec/remote_config/capabilities.js:19]: good catch

File: packages/dd-trace/src/config.js
Comment (simon-id) [inline packages/dd-trace/src/config.js:977]: also btw i just realized (should have said that in the previous comment), if `typeof value === 'string'`, then how could value.split be undefined ? we know it's a string sooo it should have split right ?

Comment (simon-id) [reply packages/dd-trace/src/config.js:977]: for example, no check is made here: 
image

Comment (ida613) [reply packages/dd-trace/src/config.js:977]: true that

File: packages/dd-trace/test/config.spec.js
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:682]:
```suggestion
      samplingRules,
```

**Review by simon-id — COMMENTED** [review]
LGTM but wait for APM review too

**Review by simon-id — DISMISSED** [review]

**Review by bengl — APPROVED** [review]

**Pull request merged**
